packages/react-instantsearch-dom-maps/src/Redo.js

```diff
@@ -1,49 +1,47 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { createClassNames, translatable } from 'react-instantsearch-dom';
-import { STATE_CONTEXT } from './Provider';
+import GeoSearchContext from './GeoSearchContext';
 import withGoogleMaps from './withGoogleMaps';
 
 const cx = createClassNames('GeoSearch');
+const RedoPropTypes = {
+  googleMapsInstance: PropTypes.object.isRequired,
+  translate: PropTypes.func.isRequired,
+};
 
-export class Redo extends Component {
-  static propTypes = {
-    googleMapsInstance: PropTypes.object.isRequired,
-    translate: PropTypes.func.isRequired,
-  };
+export const Redo = ({ googleMapsInstance, translate, context }) => (
+  <div className={cx('control')}>
+    <button
+      className={cx(
+        'redo',
+        !context.hasMapMoveSinceLastRefine && 'redo--disabled'
+      )}
+      disabled={!context.hasMapMoveSinceLastRefine}
+      onClick={() => context.refineWithInstance(googleMapsInstance)}
+    >
+      {translate('redo')}
+    </button>
+  </div>
+);
 
-  static contextTypes = {
-    [STATE_CONTEXT]: PropTypes.shape({
-      hasMapMoveSinceLastRefine: PropTypes.bool.isRequired,
-      refineWithInstance: PropTypes.func.isRequired,
-    }).isRequired,
-  };
+Redo.propTypes = {
+  ...RedoPropTypes,
+  context: PropTypes.shape({
+    hasMapMoveSinceLastRefine: PropTypes.bool.isRequired,
+    refineWithInstance: PropTypes.func.isRequired,
+  }),
+};
 
-  getStateContext() {
-    return this.context[STATE_CONTEXT];
-  }
+const RedoWrapper = props => (
+  <GeoSearchContext.Consumer>
+    {context => <Redo {...props} context={context} />}
+  </GeoSearchContext.Consumer>
+);
 
-  render() {
-    const { googleMapsInstance, translate } = this.props;
-    const {
-      hasMapMoveSinceLastRefine,
-      refineWithInstance,
-    } = this.getStateContext();
-
-    return (
-      <div className={cx('control')}>
-        <button
-          className={cx('redo', !hasMapMoveSinceLastRefine && 'redo--disabled')}
-          disabled={!hasMapMoveSinceLastRefine}
-          onClick={() => refineWithInstance(googleMapsInstance)}
-        >
-          {translate('redo')}
-        </button>
-      </div>
-    );
-  }
-}
+RedoWrapper.displayName = 'Redo';
```

Comment:
I don't think it's valuable to rename it, we have two times the same component in the tree.